src/classes/RD2_DataMigrationMapper.cls

```diff
@@ -60,22 +60,166 @@ public class RD2_DataMigrationMapper {
      * @return The updated record (or null if there is an exception?)
      */
     public npe03__Recurring_Donation__c convertToEnhancedRD() {
+        setInstallmentFields(rd);
+        setStartDate(rd);
+        setDayOfMonth(rd);
+        setStatus(rd);
+        setOtherFields(rd);
+
         return rd;
     }
 
     /**
-     * @description Generate the RDSchedule Record(s) for the Recurring Donation
-     * @return List<RecurringDonationSchedule__c> Schedule records to insert
+     * @description Sets Installment Frequency and Installment Period fields where applicable
+     * @param rd Recurring Donation in the legacy format
+     * @return void
      */
-    public List<RecurringDonationSchedule__c> getScheduleRecords() {
-        return new List<RecurringDonationSchedule__c>();
+    private void setInstallmentFields(npe03__Recurring_Donation__c rd) {
+
+        Set<String> frequencyOneInstallmentPeriods = new Set<String>{
+            System.Label.npe03.RecurringDonationInstallmentPeriodYearly,
+            System.Label.npe03.RecurringDonationInstallmentPeriodMonthly,
+            System.Label.npe03.RecurringDonationInstallmentPeriodWeekly
+        };
+
+        if (rd.npe03__Installment_Period__c == System.Label.npe03.RecurringDonationInstallmentPeriodQuarterly) {
+            rd.npe03__Installment_Period__c = RD2_Constants.INSTALLMENT_PERIOD_MONTHLY;
+            rd.InstallmentFrequency__c = 3;
+
+        } else if (frequencyOneInstallmentPeriods.contains(rd.npe03__Installment_Period__c)) {
+            if (rd.InstallmentFrequency__c == null) {
+                rd.InstallmentFrequency__c = 1;
+            }
+
+        } else if (rd.npe03__Installment_Period__c != System.Label.npe03.RecurringDonationInstallmentPeriod1stand15th) {
+            handleCustomInstallmentPeriod(rd);
+        }
+
+        if (rd.npe03__Schedule_Type__c == System.Label.npe03.RecurringDonationDivideValue
+            && rd.npe03__Installment_Amount__c != null 
+            && rd.npe03__Installment_Amount__c != 0
+        ) {
+            rd.npe03__Amount__c = rd.npe03__Installment_Amount__c;
+        }
     }
 
     /**
-     * @description Returns true if the RD record is already in RD2 format
+     * @description Sets Installment Frequency and Installment Period for custom Installment Period
+     * @param rd Recurring Donation in the legacy format
+     * @return void
      */
-    @TestVisible
-    private Boolean isConverted() {
```

Comment:
@force2b , I have removed `isConverted()` since its check is not valid. The reason is that `Status` will be populated for every new legacy Recurring Donation created after RD2 schema has been deployed. We can reintroduce `isConverted()` but we need additional/other checks. Currently when data migration batch is being executed *again* , converted RDs will be picked up and converted again.

Reply:
`isConverted()` has been added back with additional check on "Installment Frequency". Should we check for any other field?